src/svd.jl

```diff
@@ -258,25 +272,49 @@ and `ϵ` is the [`eps`](@ref) of the element type of `S`.
 !!! compat "Julia 1.12"
     The `rank(::SVD)` method requires at least Julia 1.12.
 """
-function rank(S::SVD; atol::Real = 0.0, rtol::Real = (min(size(S)...)*eps(real(float(eltype(S))))))
+function rank(S::SVD; atol::Real=0, rtol::Real = (min(size(S)...)*eps(real(float(eltype(S))))))
     tol = max(atol, rtol*S.S[1])
     count(>(tol), S.S)
 end
 
 ### SVD least squares ###
-function ldiv!(A::SVD{T}, B::AbstractVecOrMat) where T
-    m, n = size(A)
-    k = searchsortedlast(A.S, eps(real(T))*A.S[1], rev=true)
-    mul!(view(B, 1:n, :), view(A.Vt, 1:k, :)', view(A.S, 1:k) .\ (view(A.U, :, 1:k)' * _cut_B(B, 1:m)))
+"""
+    ldiv!(F::SVD, B; atol::Real=0, rtol::Real=atol>0 ? 0 : n*ϵ)
+
+Given the SVD `F` of an ``m \\times n`` matrix, multiply the first ``m`` rows of `B` in-place
+by the Moore-Penrose pseudoinverse, storing the result in the first ``n`` rows of `B`, returning `B`.
+This is equivalent to a least-squares solution (for ``m \\ge n``) or a minimum-norm solution (for ``m \\le n``).
+
+Similar to the [`pinv`](@ref) function, the solution can be regularized by truncating the SVD,
+dropping any singular values less than `max(atol, rtol*σ₁)` where `σ₁` is the largest singular value.
+The default relative tolerance is `n*ϵ`, where `n` is the size of the smallest dimension of `M`, and
+`ϵ` is the [`eps`](@ref) of the element type of `M`.
+
+!!! compat "Julia 1.13"
+    The `atol` and `rtol` arguments require Julia 1.13 or later.
+"""
+function ldiv!(F::SVD{T}, B::AbstractVecOrMat; atol::Real=0, rtol::Real = (eps(real(float(oneunit(T))))*min(size(F)...))*iszero(atol)) where T
+    m, n = size(F)
+    k = _count_svdvals(F.S, atol, rtol)
+    if k == 0
+        B[1:n, :] .= 0
+    else
+        mul!(view(B, 1:n, :), view(F.Vt, 1:k, :)', view(F.S, 1:k) .\ (view(F.U, :, 1:k)' * _cut_B(B, 1:m)))
```

Comment:
Not sure it's helpful, but we might consider reducing allocations a bit? Like

```suggestion
        temp = view(F.U, :, 1:k)' * _cut_B(B, 1:m)
        ldiv!(Diagonal(view(F.S, 1:k)), temp)
        mul!(view(B, 1:n, :), view(F.Vt, 1:k, :)', temp)
```